### PR TITLE
Validating the default values given in a variable definition (including structs)

### PR DESCRIPTION
**Pull request by NatashaWso2**
Fixed issue : https://github.com/ballerinalang/composer/issues/2357

### CODE REVIEW

File: modules/web/js/ballerina/components/struct-definition.jsx [hunk 1]
```diff
@@ -145,22 +145,43 @@ class StructDefinition extends React.Component {
...
         const bBox = { x: elementBBox.x, y: elementBBox.y, h: elementBBox.height, w: elementBBox.width };
         const self = this;
+        let state = false;
         this.renderTextBox(textValue, bBox, (value) => {
-            value = this.addQuotesToString(type, value);
-            if (!value) {
+            // Only checks for the simple literals
+            if (type === 'int' && /^[-]?\d+$/.test(value)) {
+                state = true;
+            } else if (type === 'float' && parseFloat(value)) {
```
Comment (manuranga) [inline modules/web/js/ballerina/components/struct-definition.jsx]: this check will fail when value is 0.0, which should be a valid input.

File: modules/web/js/ballerina/ast/service-definition.js [hunk 2]
```diff
@@ -154,24 +154,30 @@ class ServiceDefinition extends ASTNode {
             if (!_.isNil(assignedValue) && !_.isEmpty(assignedValue)) {
                 stmtString += ' = ' + assignedValue;
             }
-            newVariableDefinitionStatement.setStatementFromString(stmtString);
+            // newVariableDefinitionStatement.setStatementFromString(stmtString);
```
Comment (nadeeshaan) [inline modules/web/js/ballerina/ast/service-definition.js]: remove the commented codes

File: modules/web/js/ballerina/components/struct-definition.jsx (hunk 1 above)
Comment (NatashaWso2) [inline modules/web/js/ballerina/components/struct-definition.jsx]: Fixed with commit : 69b241d

File: modules/web/js/ballerina/ast/service-definition.js (hunk 2 above)
Comment (NatashaWso2) [inline modules/web/js/ballerina/ast/service-definition.js]: Fixed with commit : 69b241d

**Pull request closed without merging**